### PR TITLE
util/threadpool: cleanup

### PR DESCRIPTION
**Pull request by BusyJay**
- clean up code
- add a default context implementation, which will be used to help replacing all other threadpool in TiKV
- adjust poll strategy, so it won't wake up too frequently

### CODE REVIEW

File: src/util/threadpool.rs [hunk 1]
```diff
@@ -196,20 +251,24 @@ where
...
-        let (mut q, _) = cvar.wait_timeout(task_queue, timeout).unwrap();
+        let mut q = match timeout {
+            Some(t) => cvar.wait_timeout(task_queue, t).unwrap().0,
+            None => cvar.wait(task_queue).unwrap(),
+        };
         q.pop()
     }
 
     fn run(&mut self) {
+        let mut timeout = Some(Duration::from_secs(1));
```
Comment (siddontang) [inline src/util/threadpool.rs]: define a constant for 1, maybe we can still use a bigger value here.

**Comment by siddontang** [comment]
Rest LGTM

File: src/bin/tikv-ctl.rs [hunk 2]
```diff
@@ -249,8 +250,12 @@ fn main() {
     };
     let db_path = matches.value_of("db").unwrap();
     let db = util::rocksdb::open(db_path, ALL_CFS).unwrap();
-    let raft_db_path = matches.value_of("raftdb").unwrap();
-    let raft_db = util::rocksdb::open(raft_db_path, &[]).unwrap();
+    let raft_db = if let Some(raftdb_path) = matches.value_of("raftdb") {
```
Comment (overvenus) [inline src/bin/tikv-ctl.rs]: Irrelevant changes.

Comment (BusyJay) [reply src/bin/tikv-ctl.rs]: I think this is a bug of Github.

**Comment by BusyJay** [comment]
PTAL

**Comment by overvenus** [comment]
LGTM

**Pull request merged**